### ISSUE TITLE
[Feature Request]: Authentication using OAuth

### ISSUE THREAD

**Issue by diogoabcosta89** [Farfetch/kafkaflow#417]
### Is your request related to a problem you have?

Yes, it is.
I can't use KafkaFlow with OAuth.

SaslMechanism value for OAuthBearer is missing.
Also, requesting for an exposing a delegate for SetOAuthBearerTokenRefreshHandler().

### Describe the solution you'd like

Add those options to the KafkaFlow abstraction layer.
Options inside "SecurityInformation", used in "cluster -> WithSecurityInformation()".

It would be a nice to have if it could initialize the token (not requested at all, I only need to pass the delegate for refreshing the token).

### Are you able to help bring it to life and contribute with a Pull Request?

No

**Comment by mnrTrifork** [comment]
I'm also interested in this - At first glance, the most the complex part of this would be implementing the delegate for `SetOAuthBearerTokenRefreshHandler` in a user-friendly way.

**Comment by marcinreszko** [comment]
Any progress on this? Would love to be able to use OAuth authentication.

**Comment by robertcoltheart** [comment]
The PR/s above are ready and waiting to be merged. Waiting on a maintainer to review.
